src/database/postgres.ts

```diff
@@ -73,8 +78,30 @@ export class DatabaseWrapperPostgres implements DatabaseWrapper {
 		}
 	}
 
-	async getTournament(tournamentId: string): Promise<DatabaseTournament> {
-		return this.wrap(await this.findTournament(tournamentId));
+	// TODO: remove and replace with alternate implementation
+	// This causes many unnecessary double queries
+	public async authenticateHost(tournamentId: string, hostId: string): Promise<void> {
+		const tournament = await this.findTournament(tournamentId);
+		if (!tournament.hosts.includes(hostId)) {
+			throw new UnauthorisedHostError(hostId, tournamentId);
+		}
+	}
+
+	// TODO: remove and replace with alternate implementation
+	// This causes many unnecessary double queries
+	public async authenticatePlayer(tournamentId: string, playerId: string): Promise<void> {
+		const tournament = await this.findTournament(tournamentId);
+		if (!tournament.confirmed.find(participant => participant.discordId === playerId)) {
+			throw new UnauthorisedPlayerError(playerId, tournamentId);
+		}
+	}
+
+	async getTournament(tournamentId: string, assertStatus?: TournamentStatus): Promise<DatabaseTournament> {
+		const tournament = await this.findTournament(tournamentId);
+		if (assertStatus && tournament.status !== assertStatus) {
+			throw new AssertStatusError(tournamentId, assertStatus, tournament.status);
+		}
+		return this.wrap(tournament);
 	}
 
 	async updateTournament(tournamentId: string, name: string, desc: string): Promise<void> {
```

Comment:
Worth changing the thrown exception to AssertStatusError here too?

Reply:
Yes, I think updateTournament should use the same error